### PR TITLE
Jest: Add .babelrc.js workaround to fix snapshot issues with v28

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

Since the [`jest@28.x` update](https://github.com/strapi/strapi/pull/13740) we are running into an issue, where updating the snapshots doesn't work any longer, if jest doesn't find the babel configuration, to transform jsx. The [tracking issue in jest ](https://github.com/facebook/jest/issues/11741) is open since mid 2021 and it looks like as if nothing changes very soon (the bug was introduced in jest@27).

The only known and working solution is to add/ rename the current babel configurations to `.babelrc.js`. As a temporary workaround until the issue is fixed I'd propose we add the missing `.babelrc.js` files to each package. I've also pulled out some of the configuration of `admin` from webpack into the babel configuration, where it imo belongs.

In addition, I've removed some transforms which are not needed anylonger, since we are using @babel/preset-env.

Alternative considered: We can not downgrade jest, because we depend on some new features.

### Why is it needed?

At the moment we are running into an issue where a snapshot test can not be updated once it is located in a package. Jest [will complain that `@babel/preset-react` is not found and fail to update](https://github.com/strapi/strapi/pull/13842#issuecomment-1193975474) the snapshot.

### Related issue(s)/PR(s)

- Refs https://github.com/facebook/jest/issues/11741

### CODE REVIEW

File: packages/core/helper-plugin/babel.config.js [hunk 1]
```diff
@@ -1,21 +0,0 @@
...
-    },
-  ],
-];
-
-module.exports = {
-  presets: defaultPresets.concat(['@babel/preset-react', '@babel/preset-flow']),
-  plugins: [
-    ['@babel/plugin-proposal-class-properties', { loose: true }],
-    ['@babel/plugin-proposal-private-property-in-object', { loose: true }],
-    ['@babel/plugin-proposal-private-methods', { loose: true }],
```
Comment (Convly) [inline packages/core/helper-plugin/babel.config.js:16]: I guess those are the plugins that are no longer needed and that you mentioned in your PR description?

Comment (gu-stav) [reply packages/core/helper-plugin/babel.config.js:16]: Yes. They are all part of `@babel/preset-env` now.

**Comment by ronronscelestes** [comment]
Tested on a branch with the updated DS - it seems to have fixed the issue of not being able to update snapshots, thank you 🙏🔥

**Review by Convly — APPROVED** [review]
LGTM

**Pull request closed without merging**